test/router-luatest/router_test.lua

```diff
@@ -444,3 +444,113 @@ g.test_ssl = function(g)
     vtest.router_cfg(g.router, cluster_cfg)
     vtest.storage_wait_fullsync(g)
 end
+
+g.test_enable_disable = function(g)
+    --
+    -- gh-291: router enable/disable
+    --
+    local router = vtest.router_new(g, 'router_1')
+    -- do not allow router's configuration to complete
+    router:exec(function()
+        _G.ivshard.router.internal.errinj.ERRINJ_CFG_DELAY = true
+    end)
+    router:exec(function(cfg)
+        rawset(_G, 'fiber_static', ifiber.create(ivshard.router.cfg, cfg))
+        rawset(_G, 'fiber_new', ifiber.create(ivshard.router.new,
+                                              'new_router', cfg))
+        _G.fiber_static:set_joinable(true)
+        _G.fiber_new:set_joinable(true)
+    end, {cluster_cfg})
+
+    -- emulate unconfigured box
+    router:exec(function()
+        rawset(_G, 'old', box.cfg)
+        box.cfg = function(...) return _G.old(...) end
+    end)
+
+    -- check whether errors are not nil and their messages are equal to str
+    local assert_errors_equals = function(err1, err2, str)
+        t.assert(err1 and err2)
+        t.assert_equals(err1.message, err2.message)
+        t.assert_str_contains(err1.message, str)
+    end
+
+    local err1, err2 = router:exec(function()
+        rawset(_G, 'static_router', ivshard.router.internal.routers._static_router)
+        rawset(_G, 'new_router', ivshard.router.internal.routers.new_router)
+        local _, err_1 = pcall(_G.static_router.info, _G.static_router)
+        local _, err_2 = pcall(_G.new_router.info, _G.new_router)
+        return err_1, err_2
+    end)
+    assert_errors_equals(err1, err2, 'box seems not to be configured')
+
+    -- set box status to loading
+    router:exec(function()
+        box.cfg = _G.old
+        _G.old = box.info
```

Comment:
Lets give a bit more meaningful names. For example, `_G.old_box_info`, `_G.old_box_cfg`.